var/spack/repos/builtin/packages/hydrogen/package.py

```diff
@@ -107,14 +106,14 @@ def libs(self):
             'libEl', root=self.prefix, shared=shared, recursive=True
         )
 
-    @when('@:0.84' or '@0.99:')
+    @when('@:0.98.0')
     def cmake_args(self):
-        spec = self.spec
+        raise InstallError("Hydrogen did not exist before v0.99." +
+                           "Did you mean to use Elemental instead?")
```

Comment:
Can we replace this with a conflicts statement? Something like:
```python
conflicts('@0:0.98', msg='Hydrogen did not exist before v0.99. ' +
                         'Did you mean to use Elemental instead?')
```
The lower bound of 0 allows your `@local` to work. The string was also missing a space after the period.

The benefit of `conflicts` over wrapping this in `cmake_args` is:

1. It makes more logical sense. This has nothing to do with `cmake`.
2. It fails at concretization time, not installation time. This means that it fails immediately, not after it finishes installing all of the dependencies.

This also allows you to drop the `@when` below.

Reply:
Didn't know you could specify conflicts like that